wodles/aws/aws_s3.py

```diff
@@ -715,17 +749,34 @@ def build_s3_filter_args(self, aws_account_id, aws_region, iterating=False, cust
         if self.reparse:
             if self.only_logs_after:
                 filter_marker = self.marker_only_logs_after(aws_region, aws_account_id)
+            else:
+                query_first_key = self.db_connector.execute(
+                    self.sql_find_first_key_processed.format(
+                        bucket_path=self.bucket_path,
+                        table_name=self.db_table_name,
+                        aws_account_id=aws_account_id,
+                        aws_region=aws_region,
+                        prefix=self.prefix
+                    )
+                )
+                try:
+                    filter_marker = query_first_key.fetchone()[0]
+                except (TypeError, IndexError):
+                    # The DB is empty and there's no only_logs_after, we start from the default date
+                    filter_marker = self.marker_custom_date(aws_region, aws_account_id, self.default_date)
```

Comment:
If no only_logs_after value was specified along with the reparse option we should use the `datetime.utcnow().replace(hour=0, minute=0, second=0, microsecond=0)`  instead of reparsing from the very first processed file in the database for consistency with the other modules and the only_logs_after parameter.